boards/esp32s3-devkit/Kconfig

```diff
@@ -0,0 +1,59 @@
+# Copyright (c) 2020 HAW Hamburg
+#               2022 Gunar Schorcht
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+config BOARD
+    default "esp32s3-devkit" if BOARD_ESP32S3_DEVKIT
+
+config BOARD_ESP32S3_DEVKIT
+    bool
+    default y
+    select BOARD_COMMON_ESP32S3
+    select CPU_MODEL_ESP32S3_WROOM_1X_N8 if BOARD_VERSION_ESP32S3_DEVKITC_1_N8
+    select CPU_MODEL_ESP32S3_WROOM_1X_N8R2 if BOARD_VERSION_ESP32S3_DEVKITC_1_N8R2
+    select CPU_MODEL_ESP32S3_WROOM_1X_N8R8 if BOARD_VERSION_ESP32S3_DEVKITC_1_N8R8
+    select CPU_MODEL_ESP32S3_WROOM_1X_N16R8V if BOARD_VERSION_ESP32S3_DEVKITC_1_N16R8V
+    select CPU_MODEL_ESP32S3_WROOM_1X_N32R8V if BOARD_VERSION_ESP32S3_DEVKITC_1_N32R8V
+    select CPU_MODEL_ESP32S3_WROOM_1X_N8 if BOARD_VERSION_ESP32S3_DEVKITC_1U_N8
+    select CPU_MODEL_ESP32S3_WROOM_1X_N8R2 if BOARD_VERSION_ESP32S3_DEVKITC_1U_N8R2
+    select CPU_MODEL_ESP32S3_WROOM_1X_N8R8 if BOARD_VERSION_ESP32S3_DEVKITC_1U_N8R8
+    select CPU_MODEL_ESP32S3_MINI_1X_N8 if BOARD_VERSION_ESP32S3_DEVKITM_1_N8
+    select CPU_MODEL_ESP32S3_MINI_1X_N8 if BOARD_VERSION_ESP32S3_DEVKITM_1U_N8
+    select HAS_ARDUINO
+    select HAS_ESP_JTAG
+    select HAS_PERIPH_ADC
+    select HAS_PERIPH_I2C
+    select HAS_PERIPH_PWM
+    select HAS_PERIPH_SPI
+
+    choice
+        bool "ESP32-S3-DevKit board version"
+        depends on TEST_KCONFIG
```

Comment:
Why does this have to depend on `TEST_KCONFIG`? When `TEST_KCONFIG=n` there is no CPU selected, right?

Reply:
Because it should only be visible if other ESP32 configurations are visible. Otherwise you would see the board version selection as the only ESP stuff which seems a little bit strange.

Reply:
> Because it should only be visible if other ESP32 configurations are visible. Otherwise you would see the board version selection as the only ESP stuff which seems a little bit strange.

But `TEST_KCONFIG` is meant to be a switch to indicate that dependency resolution is handled with Kconfig, thus reserved for `MODULE_` Kconfig symbols.

Reply:
I see, I didn't know that.

Reply:
Hmm I was able to reproduce the behavior.  It seems something is up when menuconfig is used with `TEST_KCONFIG=1` I will look into it and see if it on master.

Reply:
Just to note this behaviour seems to exist only in the esp boards with variants (though testing was not extensive).
We get more than 2 `=== [ATTENTION] Testing Kconfig dependency modelling ===` warning which means that there is there are multiple calls, probably due to this, the kconfig settings are being overwritten with defaults.

Further investigation required.

Reply:
FYI, this problem exists elsewhere.

Reply:
Yup turns out that one must clean before toggling the TEST_KCONFIG...  Please try to reproduce the issue after cleaning.